go/worker/storage/committee/node.go

```diff
@@ -665,21 +674,32 @@ func (n *Node) updateExternalServicePolicy(rtComputeNodes nodes.NodeDescriptorLo
 
 	executorCommitteePolicy.AddRulesForNodeRoles(&policy, rtComputeNodes.GetNodes(), node.RoleComputeWorker)
 
-	// TODO: Query registry only for storage nodes after
-	// https://github.com/oasisprotocol/oasis-core/issues/1923 is implemented.
-	nodes, err := n.commonNode.Consensus.Registry().GetNodes(n.ctx, consensus.HeightLatest)
-	if err != nil {
-		n.logger.Error("couldn't get nodes from registry", "err", err)
-	}
-	if len(nodes) > 0 {
-		// Only include storage nodes for our runtime.
-		var storageNodes []*node.Node
-		for _, nd := range nodes {
-			if nd.GetRuntime(n.commonNode.Runtime.ID()) != nil && nd.HasRoles(node.RoleStorageWorker) {
-				storageNodes = append(storageNodes, nd)
+	switch {
+	// If public storage RPC was enabled in the config, then the normally gated methods need to be allowed
+	// for everyone.
+	case n.rpcRoleProvider != nil:
+		for _, act := range storageNodesPolicy.Actions {
+			policy.AllowAll(act)
+		}
+
+	// If not configured otherwise, state access should be restricted to storage committee members.
+	default:
```

Comment:
Why is this part not covered? Are we not testing with any storage nodes that don't expose public RPC? We should test those as well (e.g., configure a storage node without public RPC exposed). Maybe in cases where we have two storage nodes, set one to expose public RPC and the other to not expose it?